src/Entity/Enemy.cpp

```diff
@@ -47,76 +67,187 @@ void InitEnemy(flecs::world& ecs)
+	const Texture2D mapTexture = ecs.component<MapTag>().target<MapTag>().get_ref<TextureComponent>()->mTexture;
 
-			if (CheckCollisionRecs(newGoblinRect, otherRect))
-			{
-				hasCollided = true;
-				return;
-			}
-		}
-	});
+	switch (cornerIndex)
+	{
+		case 0:
+			spawnPosition = {0.0f + gCharacterSize.x, 0.0f + gCharacterSize.y};
+			break;
+		case 1:
+			spawnPosition = {0.0f + gCharacterSize.x, (float)mapTexture.height - gCharacterSize.y};
+			break;
+		case 2:
+			spawnPosition = {(float)mapTexture.width - gCharacterSize.x, 0.0f + gCharacterSize.y};
+			break;
+		default:
+			spawnPosition = {(float)mapTexture.width - gCharacterSize.x, (float)mapTexture.height - gCharacterSize.y};
+			break;
+	}
+	
+	const TransformComponent transformComponent{spawnPosition, gCharacterSize};
+	const Rectangle newGoblinRect{transformComponent.mPosition.x, transformComponent.mPosition.y, gCharacterSize.x, gCharacterSize.y};
 
+	//Picking a tower
+	TargetTowerComponent targetTowerComponent = PickTowerTarget(ecs, transformComponent);
+
+	bool hasCollided = physicsUtils::RectCollision(ecs, newGoblinRect);
 	if (hasCollided)
 		return;
 
+	//Setting the name of the goblin
 	static int index = 0;
 	const std::string goblinName = std::string("goblin" + std::to_string(index++));
+		
 	ecs.entity(goblinName.c_str())
 			.is_a(gGoblinPrefab)
+			.set<TargetTowerComponent>(targetTowerComponent)
 			.set<TransformComponent>(transformComponent);
+}
+
+TargetTowerComponent PickTowerTarget(flecs::world& ecs, const TransformComponent& enemyTransform)
+{
+	auto filter = ecs.filter<TowerTag, TransformComponent>();
+	flecs::entity_t closestTowerEntity = NULL; //QUESTION: NULL or nullptr?
 	
-	//TODO: If i want to add an target entity: .emplace<TargetComponent>(ecs.component<CharacterTag>().target<CharacterTag>().get_ref<TransformComponent>())
+	// Initialize to the maximum possible float value
+	float minDistance = std::numeric_limits<float>::max();
+
+	//QUESTION: each didnt work on entity_t?
+	//filter.each([&](flecs::entity_t towerEntityId) 
+	//QUESTION: couldn't get lamba to work without adding
+	//towerTags, cause it's in a filter?
+    
+	filter.iter([&](flecs::iter& it, TowerTag* towerTags, TransformComponent* towerTransformComponents) {
+		for (auto i : it)
+		{
+			// Calculate the distance from this tower to the enemy
+			float dx = towerTransformComponents[i].mPosition.x - enemyTransform.mPosition.x;
+			float dy = towerTransformComponents[i].mPosition.y - enemyTransform.mPosition.y;
+			float distance = sqrt(dx * dx + dy * dy);
+
+			// If this tower is closer than the current closest tower, update the closest tower
+			if (distance < minDistance)
+			{
+				minDistance = distance;
+				closestTowerEntity = it.entity(i).id();
+			}
+		}
+	});
+
+	return {closestTowerEntity};
 }
 
-void EnemyUpdate(flecs::entity enemyEntity, TransformComponent& transformComponent, VelocityComponent& velocityComponent)
+TargetTowerComponent PickTowerTarget(flecs::world& ecs)
```

Comment:
I’d rename to `GetRandomTowerId` for better clarity and change the return type `flecs::entity_t`